pkg/controller/machineautoscaler/machineautoscaler_controller.go

```diff
@@ -228,8 +234,8 @@ func (r *Reconciler) GetTarget(ref *corev1.ObjectReference) (*MachineTarget, err
 	obj := &unstructured.Unstructured{}
 	gvk := ref.GroupVersionKind()
 
-	if !SupportedTarget(gvk) {
-		return nil, ErrUnsupportedTarget
+	if valid, err := ValidateReference(ref); !valid || err != nil {
```

Comment:
Is it more prudent to first look at `err`, then `!valid`.

Reply:
Yeah, this is a little weird. I've updated the comment for the validation function to reflect the assumptions about returning non-nil errors. I also updated the test to ensure we always return a non-nil error for invalid references.